plugins/services/src/js/components/forms/EnvironmentFormSection.js

```diff
@@ -0,0 +1,220 @@
+      return (
+        <div key={key} className="flex row">
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.env[key])}>
+            {keyLabel}
+            <FieldInput
+              name={`env.${key}.key`}
+              type="text"
+              value={env.key}/>
+            <FieldError>{errors.env[key]}</FieldError>
+            <span className="emphasis"
+              style={{
+                position: 'absolute',
+                left: '100%',
+                bottom: '0.8em'
+              }}>:</span>
+          </FormGroup>
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.env[key])}>
+            {valueLabel}
+            <FieldInput
+              name={`env.${key}.value`}
+              type="text"
+              value={env.value}/>
+            <FieldError>{errors.env[key]}</FieldError>
+          </FormGroup>
+          <div className="form-group flex flex-item-align-end column-4">
+            <a className="column-3 button button-primary-link button-flush"
+              onClick={(event) => {
+                event.preventDefault();
+                this.props.onRemoveItem({value: key, path: 'env'});
+              }}>
+              Delete
+            </a>
+          </div>
+        </div>
+    ); });
+  }
+
+  getLabelsLines(data) {
+    const {errors} = this.props;
+    return data.map((label, key) => {
+      let keyLabel = null;
+      let valueLabel = null;
+      if (key === 0) {
+        keyLabel = (
+          <FieldLabel>
+            Key
+          </FieldLabel>
+        );
+        valueLabel = (
+          <FieldLabel>
+            Value
+          </FieldLabel>
+        );
+      }
+      return (
+        <div key={key} className="flex row">
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.labels[key])}>
+            {keyLabel}
+            <FieldInput
+              name={`labels.${key}.key`}
+              type="text"
+              value={label.key}/>
+            <span className="emphasis"
+              style={{
+                position: 'absolute',
+                left: '100%',
+                bottom: '0.8em'
+              }}>:</span>
+            <FieldError>{errors.labels[key]}</FieldError>
+          </FormGroup>
+          <FormGroup
+            className="column-3"
+            required={false}
+            showError={Boolean(errors.labels[key])}>
+            {valueLabel}
+            <FieldInput
+              name={`labels.${key}.value`}
+              type="text"
+              value={label.value}/>
+            <FieldError>{errors.labels[key]}</FieldError>
+          </FormGroup>
+          <div className="form-group flex flex-item-align-end column-4">
+            <a className="column-3 button button-primary-link button-flush"
+              onClick={(event) => {
```

Comment:
Let's create a `handleRemoveItem` function to separate view declaration and logic. Also, so we are not passing down a new function on each render